src/coprocessor/dag/expr/builtin_time.rs

```diff
@@ -322,6 +322,71 @@ impl ScalarFunc {
         Ok(None)
     }
 
+    fn period_to_month(period: u64) -> u64 {
+        if period == 0 {
+            return 0;
+        }
+        let (year, month) = (period / 100, period % 100);
+        if year < 70 {
+            year * 12 + month - 1 + 24000 // 2000*12
+        } else if year < 100 {
+            year * 12 + month - 1 + 22800 // 1900*12
+        } else {
+            year * 12 + month - 1
+        }
+    }
+
+    fn month_to_period(month: u64) -> u64 {
+        // maybe overflow
+        if month == 0 {
+            return 0;
+        }
+        let year = month / 12;
+        const FINAL_PERIOD: u64 = 206812;
```

Comment:
Sorry, I don't understand the`206812`.

I execute below SQL In MySQL:
```sql
mysql > select period_add(200212, 9223372036854775807)
+-----------------------------------------+
| period_add(200212, 9223372036854775807) |
+-----------------------------------------+
| 200211                                  |
+-----------------------------------------+
1 row in set
Time: 0.006s
mysql > select period_add(200212, 9223372036854775808)
+-----------------------------------------+
| period_add(200212, 9223372036854775808) |
+-----------------------------------------+
| 200212                                  |
+-----------------------------------------+
1 row in set
Time: 0.006s
mysql > select period_add(200212, 9223372036854775809)
+-----------------------------------------+
| period_add(200212, 9223372036854775809) |
+-----------------------------------------+
| 200301                                  |
+-----------------------------------------+
```

Reply:
Maybe we can refer to MySQL : https://github.com/mysql/mysql-server/blob/8.0/sql/item_timefunc.cc#L956